### PR TITLE
README.ES.md

### PR DESCRIPTION
**Pull request by zanezhub**
# Changes
I added a spanish translation for the README.md file, and modified the README.md to add my github profile and to add the link to README.ES.md file

# Translation
In the future i'd like to add more spanish translations to all the tutorials that this repository currently has.

### CODE REVIEW

File: README.ES.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+<br/>
+
+<img src="doc/header.jpg" height="372"> <img src="doc/minipush_demo_frontpage.gif" height="372">
+
+## ℹ️ Introducción
+
+Esto es una serie de tutoriales para los desarrolladores aficionados a los Sistemas Operativos (OS) 
+que se están adentrando a la nueva arquitectura ARM de 64 bits [ARMv8-A
+architecture]. Los tutoriales darán una guía paso a paso en cómo escribir un Sistema Operativo 
+[monolitico] desde cero.
```
Comment (eldruin) [inline README.ES.md:14]:
```suggestion
[monolítico][monolitico] desde cero.
```

File: README.ES.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+[ARMv8-A architecture]: https://developer.arm.com/products/architecture/cpu-architecture/a-profile/docs
+[monolitico]: https://en.wikipedia.org/wiki/Monolithic_kernel
+[@andre-richter]: https://github.com/andre-richter
+[@colachg]: https://github.com/colachg
+[@readlnh]: https://github.com/readlnh
+[@zanezhub]: https://github.com/zanezhub
+
+## 📑 Estructura
+
+- Cada tutorial contienen un solo binario booteable de la `kernel`.
```
Comment (eldruin) [inline README.ES.md:39]:
```suggestion
- Cada tutorial contiene un solo binario arrancable correspondiente al núcleo.
```

File: README.ES.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+- Cada tutorial nuevo extiende el tutorial anterior.
+- Cada tutorial tendrá un `README` y cada `README` tendrá un pequeña sección de [`tl;dr`](https://es.wikipedia.org/wiki/TL;DR) 
+  en donde se dará una pequeña recapitulación de las adiciones anteriores y se mostrará el código fuente `diff` del tutorial 
+  anterior para que se pueda inspeccionar los cambios/adiciones que han ocurrido.
```
Comment (eldruin) [inline README.ES.md:43]:
```suggestion
  anterior para que se puedan inspeccionar los cambios/adiciones que han ocurrido.
```

File: README.ES.md [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+    - Algunos tutoriales además de tener un `tl;dr` también tendrán una sección en la que se dará una explicación con lujo de detalle.
```
Comment (eldruin) [inline README.ES.md:44]:
```suggestion
    - Algunos tutoriales además de tener un `tl;dr` también tendrán una sección en la que se dará una explicación con todo lujo de detalle.
```

File: README.ES.md [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+      El plan a largo plazo es que cada tutorial tenga una buena explicación en adición al `tl;dr` y al `diff`; pero por el momento los únicos tutoriales
```
Comment (eldruin) [inline README.ES.md:45]:
```suggestion
      El plan a largo plazo es que cada tutorial tenga una buena explicación además del `tl;dr` y el `diff`; pero por el momento los únicos tutoriales
```

File: README.ES.md [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+      que gozan de una son los tutoriales en los que creo que el `tl;dr` y el `diff` no son suficientes para comprender lo que está pasando.
+- El código que se escribió en este tutorial soporta y corre en la **Raspberry Pi 3** y en la **Raspberry 4**
+  - Del tutorial 1 hasta el 5 son tutoriales "preparatorios", por lo que este código solo tendrá sentido ejecutarlo en [`QEMU`](https://www.qemu.org/).
+  - Cuando llegues al [tutorial 5](05_drivers_gpio_uart) podr comenzar a cargar y a ejecutar el kernel en una
```
Comment (eldruin) [inline README.ES.md:49]:
```suggestion
  - Cuando llegues al [tutorial 5](05_drivers_gpio_uart) podrás comenzar a cargar y a ejecutar el núcleo (kernel) en una
```

File: README.ES.md [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+    Raspeberry de verdad, y observar el output en `UART`.
```
Comment (eldruin) [inline README.ES.md:50]:
```suggestion
    Raspeberry Pi de verdad, y observar la salida serie (`UART`).
```

File: README.ES.md [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+- Aunque la Raspberry Pi 3 y 4 son las principales tarjetas este código está escrito en un estilo modular,
```
Comment (eldruin) [inline README.ES.md:51]:
```suggestion
- Aunque la Raspberry Pi 3 y 4 son las principales placas, este código está escrito en un estilo modular,
```

File: README.ES.md [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+  lo que permite una fácil portabilidad a otra arquitecturas de CPU o/y tarjetas.
```
Comment (eldruin) [inline README.ES.md:52]:
```suggestion
  lo que permite una fácil portabilidad a otras arquitecturas de CPU y/o placas.
```

File: README.ES.md [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+  - Me encantaría si alguien intenta implementar este código en una arquitectura **RISC-V**.
```
Comment (eldruin) [inline README.ES.md:53]:
```suggestion
  - Me encantaría si alguien intentase adaptar este código a una arquitectura **RISC-V**.
```

File: README.ES.md [hunk 11]
```diff
@@ -0,0 +1,168 @@
...
+- Para la edición recomiendo [Visual Studio Code] con [Rust Analyzer].
+- En adición al texto que aparece en los tutoriales también sería recomendable de revisar 
+  el comando `make doc` en cada tutorial. Este comando te deja navegar el código documentado de una manera cómoda.
+
+### Output del comando `make doc`
```
Comment (eldruin) [inline README.ES.md:58]:
```suggestion
### Salida del comando `make doc`
```

File: README.ES.md [hunk 12]
```diff
@@ -0,0 +1,168 @@
...
+
+![make doc](doc/make_doc.png)
+
+[Visual Studio Code]: https://code.visualstudio.com
+[Rust Analyzer]: https://rust-analyzer.github.io
+
+## 🛠 Requesitos del sistema
```
Comment (eldruin) [inline README.ES.md:65]:
```suggestion
## 🛠 Requisitos del sistema
```

File: README.ES.md [hunk 13]
```diff
@@ -0,0 +1,168 @@
...
+   Ejecuta esto en la carpeta root del repositorio:
+
+   ```bash
+   bundle install --path .vendor/bundle --without development
+   ```
+
+[docker group]: https://docs.docker.com/engine/install/linux-postinstall/
+[Rust Analyzer extension]: https://marketplace.visualstudio.com/items?itemName=matklad.rust-analyzer
+
+### 🧰 Más detalles: Eliminando Toolchain Hassle
```
Comment (eldruin) [inline README.ES.md:102]:
```suggestion
### 🧰 Más detalles: Eliminando Líos con Toolchains
```

File: README.ES.md [hunk 14]
```diff
@@ -0,0 +1,168 @@
...
+
+Esta serie trata de enfocarse lo máximo posible en tener una experiencia amistosa con el usario.
+Por lo tanto, se han dirigido muchos esfuerzos a eliminar la parte más difícil del desarrollo de
+los sistemas incorporados (embedded) tanto como se pudo: `Toolchain hassle`.
```
Comment (eldruin) [inline README.ES.md:106]:
```suggestion
los sistemas incorporados (embedded) tanto como se pudo.
```

File: README.ES.md [hunk 15]
```diff
@@ -0,0 +1,168 @@
...
+
+Rust por sí mismo ya está ayudando mucho, porque tiene integrado el soporte para cross-compilation.
+Todo lo que necesitamos para compilar desde una máquina con una arquitectura `x86` a una Raspberry Pi
+con arquitectura `AArch64` será automáticamente instalado por `rustup`. Sin embargo, además de usar
+el compilador de Rust, también usaremos algunas otras herrameintas, entre las cuales están:
```
Comment (eldruin) [inline README.ES.md:111]:
```suggestion
el compilador de Rust, también usaremos algunas otras herramientas, entre las cuales están:
```

File: README.ES.md [hunk 16]
```diff
@@ -0,0 +1,168 @@
...
+
+- `QEMU` para emular nuestro kernel en nuestra máquina principal.
+-  Una herramienta llamada `Minipush` para cargar el kernel en una Raspberry Pi on-demand usando `UART`.
```
Comment (eldruin) [inline README.ES.md:114]:
```suggestion
-  Una herramienta llamada `Minipush` para cargar el núcleo en una Raspberry Pi cuando queramos usando `UART`.
```

File: README.ES.md [hunk 17]
```diff
@@ -0,0 +1,168 @@
...
+- `OpenOCD` y `GDB` para hacer "debugging" de la máquina a instalar.
+
+Hay muchas cosas que pueden salir mal mientras instalamos y/o compilamos las versiones correctas de cada
+herramienta en tu máquina. Por ejemplo, tu distribución tal vez podría no proporcionar las versiones más
+recientes que se necesiten. O tal vez te falten algunas dependencias para la compilar estas herramientas.
```
Comment (eldruin) [inline README.ES.md:119]:
```suggestion
recientes de paquetes que se necesiten. O tal vez te falten algunas dependencias para la compilar estas herramientas.
```

File: README.ES.md [hunk 18]
```diff
@@ -0,0 +1,168 @@
...
+
+[install_docker]: https://docs.docker.com/get-docker/
+
+## 📟 USB Serial Output
+
+Ya que el desarrollo de este kernel se está ejecutando en hardware real, se recomienda que tengas
+un USB serial cable para sentir la experiencia completa.
+
+- Puedes encontrar estos cables que deberían funcionar sin ningún problema en [\[1\]] [\[2\]], pero
+  hay muchos otros que pueden funcionar. Idealmente, tu cable está basado en el chip `CP2102`.
```
Comment (eldruin) [inline README.ES.md:134]:
```suggestion
  hay muchos otros que pueden funcionar. Idealmente, tu cable estaría basado en el chip `CP2102`.
```

File: README.ES.md [hunk 19]
```diff
@@ -0,0 +1,168 @@
...
+- Lo conectas a los pines `GND` y GPIO `14/15` como se muestra en la parte inferior.  
+- [Tutorial 5](05_drivers_gpio_uart) es la primera vez en la que lo vas usar. Revisa las instrucciones
+  en cómo preparar una tarjeta SD para bootear en tu kernel desde ahí.
```
Comment (eldruin) [inline README.ES.md:137]:
```suggestion
  sobre cómo preparar una tarjeta SD para arrancar en tu núcleo desde ahí.
```

File: README.ES.md [hunk 20]
```diff
@@ -0,0 +1,168 @@
...
+- Empezando con el [tutorial 6](06_uart_chainloader), bootear kernels en tu Raspberry comienza a ser
```
Comment (eldruin) [inline README.ES.md:138]:
```suggestion
- Empezando con el [tutorial 6](06_uart_chainloader), arrancar núcleos en tu Raspberry Pi comienza a ser
```

File: README.ES.md [hunk 21]
```diff
@@ -0,0 +1,168 @@
...
+  más fácil. En este tutorial, un `chainloader` es desarrollado, que será el último archivo que necesitarás
```
Comment (eldruin) [inline README.ES.md:139]:
```suggestion
  más fácil. En este tutorial se desarrolla un `chainloader`, que será el último archivo que necesitarás
```

File: README.ES.md [hunk 22]
```diff
@@ -0,0 +1,168 @@
...
+  copiar de manera manual a la tarjeta SD por un tiempo. Esto te permitirá cargar los kernels de los tutoriales
```
Comment (eldruin) [inline README.ES.md:140]:
```suggestion
  copiar de manera manual a la tarjeta SD por el momento. Esto te permitirá cargar los núcleos de los tutoriales
```

File: README.ES.md [hunk 23]
```diff
@@ -0,0 +1,168 @@
...
+  durante el boot on demand usando `UART`.
```
Comment (eldruin) [inline README.ES.md:141]:
```suggestion
  durante el arranque usando `UART`.
```

File: README.ES.md [hunk 24]
```diff
@@ -0,0 +1,168 @@
...
+
+Este proyecto está licenciado por cualquiera de las siguientes licencias como alguna de tus dos opciones
+
+- Apache License, Version 2.0, ([LICENSE-APACHE](LICENSE-APACHE) or http://www.apache.org/licenses/LICENSE-2.0)
+- MIT license ([LICENSE-MIT](LICENSE-MIT) or http://opensource.org/licenses/MIT)
+
+
+### Contribución
+
+A menos de que lo menciones, cualquier contribución enviada por ti para su inclusión en este trabajó,
```
Comment (eldruin) [inline README.ES.md:164]:
```suggestion
A menos de que lo menciones, cualquier contribución enviada por ti para su inclusión en este trabajo,
```

File: README.ES.md [hunk 25]
```diff
@@ -0,0 +1,168 @@
...
+caerá bajo la licencia de Apache-2.0, deberá tener doble licencias como se muestra en la parte superior, sin ninguna
+adición de términos o condiciones.
```
Comment (eldruin) [inline README.ES.md:166]:
```suggestion
tal como se define en la licencia Apache-2.0, deberá tener doble licencia como se muestra en la parte superior, sin ningún
cambio de términos o condiciones.
```

File: README.md [hunk 26]
```diff
@@ -22,11 +22,15 @@ P.S.: Chinese :cn: versions of the tutorials were started by [@colachg] and [@re
 them as [`README.CN.md`](README.CN.md) in the respective folders. They are a bit out-of-date at the
 moment though.
 
+The [spanish version of this README.md file](README.ES.md) was created and commited by :mexico: :es: [@zanezhub].
```
Comment (eldruin) [inline README.md:25]:
```suggestion
The [Spanish version of this README.md file](README.ES.md) was created and commited by :mexico: :es: [@zanezhub].
```

File: README.md [hunk 27]
```diff
@@ -22,11 +22,15 @@ P.S.: Chinese :cn: versions of the tutorials were started by [@colachg] and [@re
...
+In the future there'll be tutorials translated in spanish.
```
Comment (eldruin) [inline README.md:26]:
```suggestion
In the future there'll be more tutorials translated into Spanish.
```

**Review by eldruin — COMMENTED** [review]
Thank you for this work!
Being a native Spanish speaker myself as well, I added a few improvements/corrections.
Other than that, looks good to me.

**Comment by andre-richter** [comment]
That's awesome, thanks!

I took the freedom and pushed over README.md to have it slightly reorganized how we present translations. You might want to adapt the spanish version accordingly 👍

**Comment by zanezhub** [comment]
@eldruin 
Thanks for the corrections and the improvements you added!

@andre-richter 
You're welcome. I already watched the README.md file you pushed on my fork and I made all the necessary changes to the spanish version. The new commit 8ce77b5 has all the improvements/corrections that @eldruin provided and all the changes that you added as well.

**Comment by andre-richter** [comment]
@eldruin can you approve once you are happy with the PR?

**Review by eldruin — APPROVED** [review]
Yes, looks good to me, thank you!

**Pull request merged**